### PR TITLE
Display of prediction samples for fastai

### PR DESCRIPTION
**Pull request by borisdayma**
This PR allows to log prediction samples when `data_type` is "images".

This should work with all vision problems treated with built-in data loaders from fastai, as well as custom datasets automatically inferred.

Samples can be given through `validation_data` or selected randomly, after which they remain the same until end of training to see the evolution of predictions.

Examples:

* classification problem: https://app.wandb.ai/borisd13/simpsons-fastai/runs/g34gpga7
* segmentation: https://app.wandb.ai/borisd13/semantic-segmentation/runs/zgf5tcu1

Using the callback does not change:

* Option 1: `learn = Learner(data, ..., callback_fns=partial(WandbCallback, data_type="images"))`
* Option 2: `learn.fit(..., callbacks=WandBCallback(learn, data_type="images"))

Once the change is merged, I'll update the documentation as listed in the module docstring.

### CODE REVIEW

File: wandb/fastai/__init__.py [hunk 1]
```diff
@@ -29,11 +29,13 @@
...
 '''
 import wandb
-import matplotlib
-matplotlib.use('Agg')  # non-interactive back-end (avoid issues with tkinter)
-import matplotlib.pyplot as plt
 from fastai.callbacks import TrackerCallback
 from pathlib import Path
+import random
+try:
+    import matplotlib.pyplot as plt
```
Comment (vanpelt) [inline wandb/fastai/__init__.py]: Do you think we should set the backend to Agg here?  I've definitely run into issues especially in docker containers where matplotlib just bombs out because we're headless.  I assume if someone wants to use a tk backend they can override our call here...

File: wandb/fastai/__init__.py [hunk 2]
```diff
@@ -43,23 +45,27 @@ class WandbCallback(TrackerCallback):
...
                  learn,
-                 log=None,
-                 show_results=False,
-                 save_model=False,
+                 log="all",
+                 save_model=True,
                  monitor='val_loss',
-                 mode='auto'):
+                 mode='auto',
+                 data_type=None,
```
Comment (vanpelt) [inline wandb/fastai/__init__.py]: We're about to change the interface of our keras callback to accept input_type and output_type.  These can often be inferred but I suggest for now you just change this to input_type.  2 different types allows us to explicitly do image classification, segmentation, etc.

File: wandb/fastai/__init__.py [hunk 3]
```diff
@@ -43,23 +45,27 @@ class WandbCallback(TrackerCallback):
 
     def __init__(self,
                  learn,
-                 log=None,
-                 show_results=False,
-                 save_model=False,
+                 log="all",
```
Comment (vanpelt) [inline wandb/fastai/__init__.py]: I think we should set this to gradients by default like `wandb.watch`

File: wandb/fastai/__init__.py (hunk 1 above)
Comment (borisdayma) [reply wandb/fastai/__init__.py]: Good point, I had it before and removed the dependency of matplotlib at some point until I needed it again. I forgot to put back agg backend

File: wandb/fastai/__init__.py (hunk 2 above)
Comment (borisdayma) [reply wandb/fastai/__init__.py]: Ok, let me know once it's revised so I try to replicate it. In fastai we can often get type of data from the name of the dataset class (does not work with custom classes).

File: wandb/fastai/__init__.py (hunk 3 above)
Comment (borisdayma) [reply wandb/fastai/__init__.py]: Is there a specific reason behind it? I would assume the more data we log, the better… unless it affects performance

**Comment by borisdayma** [comment]
I should have implemented the requested changes. I'm just waiting for confirmation on the last point related to `log` default value.

File: wandb/fastai/__init__.py (hunk 3 above)
Comment (vanpelt) [reply wandb/fastai/__init__.py]: We chose gradients because there’s added bandwidth and compute on the client and server for both.  Felt like the default should add the most value for least cost.
